src/laser/MultiLaser.cpp

```diff
@@ -0,0 +1,830 @@
+/* Copyright 2022
+ *
+ * This file is part of HiPACE++.
+ *
+ * Authors: MaxThevenet, AlexanderSinn
+ * Severin Diederichs, atmyers, Angel Ferran Pousa
+ * License: BSD-3-Clause-LBNL
+ */
+
+#include "MultiLaser.H"
+#include "utils/Constants.H"
+#include "Hipace.H"
+#include "utils/HipaceProfilerWrapper.H"
+#ifdef AMREX_USE_CUDA
+#  include "fields/fft_poisson_solver/fft/CuFFTUtils.H"
+#endif
+#include <AMReX_GpuComplex.H>
+
+#ifdef AMREX_USE_CUDA
+#  include <cufft.h>
+#elif defined(AMREX_USE_HIP)
+#  include <cstddef>
+#  include <rocfft.h>
+#else
+#  include <fftw3.h>
+#endif
+
+void
+MultiLaser::ReadParameters ()
+{
+    amrex::ParmParse pp("lasers");
+    getWithParser(pp, "names", m_names);
```

Comment:
Should we maybe make this `queryWithParser` and set a default name of "no_laser"?
Otherwise, we break every existing input script. Could be fine of course, but maybe we need to be clearer in the documentation about it.

Reply:
Good point. Just for laser? Or we could change it for beams, plasmas and lasers, to make them consistent.